### PR TITLE
Fix requirements check failure for servers without OS or arch requirements set

### LINKED ISSUE #1154: Server creation fails with "OS (linux) not supported" error even though template doesn't list an OS requirement
I'm trying to add some servers to a local PufferPanel dev environment so I can dig into #1149 a bit more. For now, I'm working on Linux (Debian in WSL2, to be specific) so I can experiment with behavior there and see how it differs from the behavior I observe on Windows. However, in the latest Git master, when I try create a server from one of the stock templates, creation fails with this error:

```
[GIN] 2022/04/18 - 18:27:30 | 200 |       439.7µs |             ::1 | GET      "/api/templates/minecraft-vanilla"
[ERROR] 2022/04/18 18:27:43 OS (linux) not supported. Supported OS:
[GIN] 2022/04/18 - 18:27:43 | 400 |      260.23µs |                 | PUT      "/daemon/server/bb73e45c"
[ERROR] 2022/04/18 18:27:43 Unexpected response from daemon: 400
[ERROR] 2022/04/18 18:27:43 unknown error
```

I think it's trying to apply [this requirements check](https://github.com/PufferPanel/PufferPanel/blob/42fd3c02e625da978d7d1d7bcad28a5578a72fcb/server.go#L101-L113) even though my template doesn't have a `requirements` section populated.

### PR DESCRIPTION
**Pull request by bcat**
Fixes #1154.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -185,6 +186,9 @@ func (s Server) DataToMap() map[string]interface{} {
 }
 
 func parseRequirementRow(str string) []string {
+	if str == "" {
```
Comment (LordRalex) [inline server.go:189]: Return an empty array instead, nil is special, where-as empty is not.

Comment (bcat) [reply server.go:189]: Done. (I haven't written much Go and wasn't sure if empty vs. nil slice was preferred.)

**Comment by bcat** [comment]
Oh, sorry about the reordered imports. I guess the standard vim-go plugin formats them like that by default. I can revert/try to figure out how to disable that if you want.

**Comment by LordRalex** [comment]
> Oh, sorry about the reordered imports. I guess the standard vim-go plugin formats them like that by default. I can revert/try to figure out how to disable that if you want.

I don't care enough to have you fixed it. It's not in order to start with, and an extra line isn't an issue.

**Pull request merged**